scripts/interfaces.ts

```diff
@@ -0,0 +1,167 @@
+// Native
+import { readFile, writeFile } from "fs/promises";
+import { mkdirSync } from "fs";
+import { basename, parse, dirname } from "path";
+
+// Vendor
+import glob from "glob";
+
+// Utilities
+import { createHashFromFile } from "./utilities/createFileHash";
+import { logger } from "./utilities/logger";
+import { spawnProcess } from "./utilities/spawnProcess";
+
+// TODO: investigate if there are ways we could generate these from compiled artifacts far more quickly
+// TODO: currently the generator stumbles on subsequent calls whenever an invalid ABI is generated (i.e. FusePoolDirectoryArbitrum.sol:18:51)
+// TODO: replace so we can parallelize the execution with limitations
+
+const main = async () => {
+  const IGNORE_LIST = [
+    // Directories
+    "test",
+    "external",
+    "interfaces",
+    // Files
+    "chains/arbitrum/FusePoolDirectoryArbitrum.sol",
+    "core/ComptrollerStorage.sol",
+    "core/CTokenInterfaces.sol",
+    "core/ErrorReporter.sol",
+    "core/RewardsDistributorStorage.sol",
+    "src/FusePoolLens.sol",
+    "src/FusePoolLensSecondary.sol",
+    "src/FuseSafeLiquidator.sol",
+  ];
+
+  const PROJECT_ROOT_DIR = `${__dirname}/..`;
+  const ABI_DIR = `${PROJECT_ROOT_DIR}/abi`;
+  const INTERFACES_DIR = `${PROJECT_ROOT_DIR}/src/test/interfaces`;
+  const SCRIPTS_DIR = `${PROJECT_ROOT_DIR}/scripts`;
+
+  const FILEPATHS = glob
+    .sync(`${PROJECT_ROOT_DIR}/src/**/*.sol`, {
+      nodir: true,
+    })
+    .filter(
+      (filePath) =>
+        !IGNORE_LIST.map((item) => filePath.includes(item)).includes(true)
+    );
+
+  const PREVIOUS_HASHES = JSON.parse(
+    await readFile(`${SCRIPTS_DIR}/hashes.json`, "utf-8")
+  );
+
+  const NEW_HASHES = Object.assign(
+    {},
+    ...(await Promise.all(
+      FILEPATHS.map((filePath) => createHashFromFile(filePath))
+    ))
+  );
+
+  const DIFF_HASHES = Object.entries({
+    ...PREVIOUS_HASHES,
+    ...NEW_HASHES,
+  }).reduce((acc, [key, value]) => {
+    if (
+      !Object.values(PREVIOUS_HASHES).includes(value) ||
+      !Object.values(NEW_HASHES).includes(value)
+    )
+      (acc as any)[key] = value;
+
+    return acc;
+  }, {});
+
+  // Filter filepaths down to diff hashes
+  const DIFF_HASHES_PATHS = Object.keys(DIFF_HASHES)
+    .map((item) =>
+      FILEPATHS.filter((filePath) => filePath.includes(`/${item}`))
```

Comment:
Not optimal because we iterate over all filepaths but the size of the filepaths are not large enough to spend time changing this